simtools/io_handler.py

```diff
@@ -33,7 +33,9 @@ def getOutputDirectory(filesLocation, label, mode):
     '''
     today = datetime.date.today()
     labelDir = label if label is not None else 'd-' + str(today)
-    path = Path(filesLocation).joinpath('simtools-output').joinpath(labelDir).joinpath(mode)
+    path = Path(filesLocation).joinpath('simtools-output').joinpath(labelDir)
+    if mode is not None:
```

Comment:
This was well spotted, thanks.

Originally getOutputDirectory was an internal function that I used to produce others, e.g getRayTracingOutputDirectory etc.
Each one of these would use a specific mode, so there was no need to limit the options.
It turns out that I ended up using getOutputDirectory outside io_handler because I neglected the pattern that I created.

Now I fixed it by returning to the original pattern. I changed the name to _getOutputDirectory, to make clear that this is an internal function and created other functions to be used outside, which do not require mode.

And I added the description to the mode argument.

Reply:
OK!
 I am not sure if I can see this change already (did you push it?). But I don't have to check it to this detail so I can resolve this conversation as well.